airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/AirbyteValue.kt

```diff
@@ -22,7 +22,7 @@ sealed interface AirbyteValue {
                 null -> NullValue
                 is String -> StringValue(value)
                 is Boolean -> BooleanValue(value)
-                is Int -> IntValue(value)
+                is Int -> IntegerValue(value.toLong())
```

Comment:
What's the diff between Int and Integer in this context?

Reply:
IntValue is a hack to enable int-based date types in avro.

An issue created by the bespoke AirbyteType/Value format is that the set of types supported by destinations is larger than those supported by airbyte.

For example, Avro wants specific integral time types (longs for timestamps/times, ints for dates).

Naively, you'd add it to the list of AirbyteTypes, but that forces every client of AirbyteType to handle types that aren't supported by Airbyte or relevant to their destination.

The approach we settled on was to change the data types only and just have avro expect Date/Time/stampType will all be integral.

But it's finicky about longs-v-ints, so we added IntValue as a kludge. A better approach would be to drop it and have the conversion in AirbyteValueToAvroRecord be schema-aware.